src/utils.rs

```diff
@@ -825,85 +823,114 @@ fn get_meta_info(
             return Err(Error::new(meta.span(), "Attribute format not supported1"));
         }
     };
-    for element in list.nested.into_iter() {
-        let nested_meta = if let NestedMeta::Meta(meta) = element {
-            meta
-        } else {
-            return Err(Error::new(meta.span(), "Attribute format not supported3"));
-        };
-        if let Meta::Path(_) = nested_meta {
-        } else {
-            return Err(Error::new(meta.span(), "Attribute format not supported4"));
-        }
-        let ident = if let Some(ident) =
-            nested_meta.path().segments.first().map(|p| &p.ident)
-        {
-            ident
-        } else {
-            return Err(Error::new(meta.span(), "Attribute format not supported5"));
-        };
 
-        let str_ident: &str = &ident.to_string();
-        if !allowed_attr_params.contains(&str_ident) {
-            return Err(Error::new(
-                ident.span(),
-                format!(
-                    "Attribute parameter not supported. Supported attribute parameters are: {}",
-                    allowed_attr_params.join(", ")
-                ),
-            ));
-        }
+    parse_punctuated_nested_meta(&mut info, &list.nested, allowed_attr_params, true)?;
 
-        match str_ident {
-            "ignore" => {
-                info.enabled = Some(false);
-            }
-            "forward" => {
-                info.forward = Some(true);
-            }
-            "owned" => {
-                info.owned = Some(true);
+    Ok(info)
+}
+
+fn parse_punctuated_nested_meta(
+    info: &mut MetaInfo,
+    meta: &Punctuated<NestedMeta, Token![,]>,
+    allowed_attr_params: &[&str],
+    value: bool,
+) -> Result<()> {
+    for meta in meta.iter() {
+        let meta = match meta {
+            NestedMeta::Meta(meta) => meta,
+            _ => {
+                return Err(Error::new(meta.span(), "Attribute format not supported2"))
             }
-            "ref" => {
-                info.ref_ = Some(true);
+        };
+
+        match meta {
+            Meta::List(list) if list.path.is_ident("not") => {
+                if value {
+                    parse_punctuated_nested_meta(
```

Comment:
Because this is called recursively it allows `#[error(not(not(source))]`, which is confusing as it's the same as `#[error(not(source))]`. Let's call another function here, that doesn't allow any `Meta::List`.

Reply:
It's not. `if value` conditional only allows for a single top-level "not" or fails with an error otherwise.

Reply:
Ah I see, can you add a comment explaining that check then?

Reply:
Done.